src/models/Settings.js

```diff
@@ -223,19 +227,65 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, config) {
         },
         cache: true
       },
-      // checks if there are any valid configured idps.
-      socialAuthConfigured: {
-        deps: ['configuredSocialIdps'],
-        fn: function (idps) {
-          return !_.isEmpty(idps);
+      // social auth buttons order - 'above'/'below' the primary auth form (boolean)
+      socialAuthPositionTop: {
+        deps: ['configuredSocialIdps', 'idpDisplay'],
+        fn: function (configuredSocialIdps, idpDisplay) {
+          return !!(!_.isEmpty(configuredSocialIdps) && idpDisplay.toUpperCase() === 'PRIMARY');
         },
         cache: true
       },
-      // social auth buttons order - 'above'/'below' the primary auth form (boolean)
-      socialAuthPositionTop: {
-        deps: ['socialAuthConfigured', 'idpDisplay'],
-        fn: function (socialAuthConfigured, idpDisplay) {
-          return !!(socialAuthConfigured && idpDisplay.toUpperCase() === 'PRIMARY');
+      //merges social auth and custom buttons into one array
+      configuredButtons: {
+        deps: ['configuredSocialIdps', 'customButtons'],
+        fn: function (configuredSocialIdps, customButtons) {
+          var self = this;
+          var socialAuthButtons = [];
+          if(_.isArray(configuredSocialIdps)) {
+            _.each(configuredSocialIdps, function (idp) {
+              var type = idp.type.toLowerCase();
+              var dataAttr = 'social-auth-' + type + '-button';
+              var socialAuthButton = {
+                id: idp.id,
+                type: idp.type,
+                dataAttr: dataAttr,
+                className: 'social-auth-button ' + dataAttr,
+                title: idp.title = Okta.loc('socialauth.' + type + '.label'),
+                click: function (e) {
+                  e.preventDefault();
+                  OAuth2Util.getTokens(self, {idp: idp.id});
+                }
+              };
+              socialAuthButtons.push(socialAuthButton);
+            });
+          }
+          var customConfiguredButtons = [];
+          if(_.isArray(customButtons)) {
+            _.each(customButtons, function(button) {
+              var customConfiguredButton = {
+                title: button.title,
+                className: button.className + ' default-custom-button',
+                click: button.click
+              };
+              customConfiguredButtons.push(customConfiguredButton);
+            });
+          }
+          var buttonArray = [];
+          if(!_.isEmpty(socialAuthButtons)) {
+            buttonArray = buttonArray.concat(socialAuthButtons);
```

Comment:
can we, instead of concatenating 2 arrays here, use the same array since the beginning and skip the concat part? At the end we just need to add buttons to the same array and we can return that
```
var buttons = [];
// add social buttons to _buttons_
// add custom buttons to _buttons_
return buttons;
```